internal/mocks/mockgen_internal.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+# Mockgen refuses to generate mocks for internal packages.
+# This script copies the internal directory and renames it to internalpackage.
+# That way, mockgen can generate the mock.
+# Afterwards, it corrects the import paths (replaces internalpackage back to internal).
+
+TEMP_DIR=$(mktemp -d)
+mkdir -p $TEMP_DIR/src/github.com/lucas-clemente/quic-go/internalpackage
+
+cp -r $GOPATH/src/github.com/lucas-clemente/quic-go/internal/* $TEMP_DIR/src/github.com/lucas-clemente/quic-go/internalpackage
```

Comment:
It's not required to have GOPATH set anymore, the script should work in that case :)

Reply:
You can even have multiple GOPATHs. In principle, this script should handle that as well, but I don't want to handle all those cases here. mockgen is a pain!